modules/opencensus/src/main/java/com/spotify/ffwd/opencensus/OpenCensusPluginSink.java

```diff
@@ -0,0 +1,168 @@
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+
+package com.spotify.ffwd.opencensus;
+
+import com.google.inject.Inject;
+import com.spotify.ffwd.model.Batch;
+import com.spotify.ffwd.model.Metric;
+import com.spotify.ffwd.output.FakeBatchablePluginSinkBase;
+import com.spotify.ffwd.output.PluginSink;
+import eu.toolchain.async.AsyncFramework;
+import eu.toolchain.async.AsyncFuture;
+import io.opencensus.exporter.stats.stackdriver.StackdriverStatsConfiguration;
+import io.opencensus.exporter.stats.stackdriver.StackdriverStatsExporter;
+import io.opencensus.stats.Aggregation.Sum;
+import io.opencensus.stats.Measure.MeasureLong;
+import io.opencensus.stats.Stats;
+import io.opencensus.stats.StatsRecorder;
+import io.opencensus.stats.View.Name;
+import io.opencensus.stats.View;
+import io.opencensus.tags.TagContext;
+import io.opencensus.tags.TagContextBuilder;
+import io.opencensus.tags.TagKey;
+import io.opencensus.tags.TagValue;
+import io.opencensus.tags.Tagger;
+import io.opencensus.tags.Tags;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/***
+ * PluginSink to export metrics into Stackdriver via OpenCensus API.
+ *
+ * Right now, this is specifically designed to export event counts for
+ * use in SLAs but could easily be extended to support more view types
+ * by allowing them to be specified in config.
+ *
+ * Resources can be specified using the environment variables
+ * OC_RESOURCE_TYPE and OC_RESOURCE_LABELS until the api becomes stable
+ * https://www.javadoc.io/doc/io.opencensus/opencensus-api/latest/io/opencensus/resource/
+package-summary.html
+ */
+public class OpenCensusPluginSink extends FakeBatchablePluginSinkBase implements PluginSink  {
+  private static final Logger log = LoggerFactory.getLogger(OpenCensusPluginSink.class);
+  @Inject
+  private AsyncFramework async;
+
+  private Map<String, MeasureLong> measures;
+
+  private static final Tagger tagger = Tags.getTagger();
+  private static final StatsRecorder statsRecorder = Stats.getStatsRecorder();
+
+  private Optional<String> gcpProject;
+
+  public void init() {
+    // Is this actually needed, not sure. Better safe than sorry!
+    measures = new ConcurrentHashMap();
+  }
+
+  public OpenCensusPluginSink(Optional<String> gcpProject) {
+    this.gcpProject = gcpProject;
+  }
+
+  public void sendMetric(Metric metric) {
+    try {
+      MeasureLong measure = measures.get(metric.getKey());
```

Comment:
i'm thinking the key for the measure needs to be the combination of the `key + what` in order to make it unique. `key` is used as a generic mapping of what is producing the metrics. In the example below key is used for all system metrics 

```
{"key":"system","value":0.008,"time":1579790379910,"tags":{"site":"gae","role":"heroicmetadataes","what":"cpu-usage","pod":"gae2","cpu_type":"user","host":"host1","env":"production"},"resource":{}}}
{"key":"system","value":0.0,"time":1579790379910,"tags":{"site":"gae","role":"heroicmetadataes","what":"cpu-usage","pod":"gae2","cpu_type":"nice","host":"host1","env":"production"},"resource":{}}}
{"key":"system","value":0.003,"time":1579790379911,"tags":{"site":"gae","role":"heroicmetadataes","what":"cpu-usage","pod":"gae2","cpu_type":"system","host":"host1","env":"production"},"resource":{}}}
{"key":"system","value":0.9890000000000001,"time":1579790379911,"tags":{"site":"gae","role":"heroicmetadataes","what":"cpu-usage","pod":"gae2","cpu_type":"idle","host":"host1","env":"production"},"resource":{}}}
{"key":"system","value":0.0,"time":1579790379911,"tags":{"site":"gae","role":"foo","what":"cpu-usage","pod":"gae2","cpu_type":"iowait","host":"host1","env":"production"},"resource":{}}}
{"key":"system","value":0.0,"time":1579790379912,"tags":{"site":"gae","role":"foo","what":"cpu-usage","pod":"gae2","cpu_type":"irq","host":"host1","env":"production"},"resource":{}}}
{"key":"system","value":0.001,"time":1579790379912,"tags":{"site":"gae","role":"foo","what":"cpu-usage","pod":"gae2","cpu_type":"softirq","host":"host1","env":"production"},"resource":{}}}
{"key":"system","value":0.0,"time":1579790379912,"tags":{"site":"gae","role":"foo","what":"cpu-usage","pod":"gae2","cpu_type":"steal","host":"host1","env":"production"},"resource":{}}}
{"key":"system","value":0.0,"time":1579790379912,"tags":{"site":"gae","role":"foo","what":"cpu-usage","pod":"gae2","cpu_type":"guest","host":"host1","env":"production"},"resource":{}}}
```

Reply:
Fair point, I've added a config item to control the metric name, hope this is ok? It didn't feel appropriate to hardcode the logic.